app/models/concerns/form526_rapid_ready_for_decision_concern.rb

```diff
@@ -1,10 +1,32 @@
 # frozen_string_literal: true
 
 require 'mail_automation/client'
+require 'lighthouse/veterans_health/client'
 
+# rubocop:disable Metrics/ModuleLength
 module Form526RapidReadyForDecisionConcern
   extend ActiveSupport::Concern
 
+  class AccountNotFoundError < StandardError; end
+
+  def get_icn
+    raise AccountNotFoundError, "for user_uuid: #{user_uuid} or their edipi" unless account
+
+    account.icn.presence
+  end
+
+  def account
+    account = Account.lookup_by_user_uuid(user_uuid)
+    return account if account
+
+    edipi = auth_headers['va_eauth_dodedipnid'].presence
+    Account.find_by(edipi: edipi) if edipi
+  end
```

Comment:
This looks like duplicate code from `app/workers/rapid_ready_for_decision/claim_context.rb`. Can you use `ClaimContext` instead?